markdown/ch7.markdown

```diff
@@ -342,12 +342,13 @@ def align(reference, target, cost=cost_mse):
         res = optimize.minimize(cost, p, args=(ref, tgt), method='Powell')
         p = res.x
 
-        print('Pyramid level %i' % n)
-        print('Angle:', np.rad2deg(res.x[0]))
-        print('Offset:', res.x[1:] * 2 ** n)
-        print('Cost function:', res.fun)
-        print('')
+        # print current level, overwriting each time (like a progress bar)
+        print('Level: %i, Angle: %.2f, Offset: (%.1f, %.1f), Cost: %.2f' %
```

Comment:
I think the f-strings are a huge improvement. I've never used % in my own code, only .format(). I feel lucky that I arrived on the Python scene just late enough to have never been required to use %.